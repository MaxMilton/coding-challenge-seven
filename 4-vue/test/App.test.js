'use strict';

// const Vue = require('vue');
// const { shallowMount } = require('@vue/test-utils');
const { mount } = require('@vue/test-utils');
const { createRenderer } = require('vue-server-renderer');
const App = require('../src/App.vue');

describe('App root component', () => {
  it('renders correctly', () => {
    // console.log('!! App', App);
    // expect.assertions(1);
    const renderer = createRenderer();
    // const wrapper = shallowMount(App);
    const wrapper = mount(App);
    // console.log('!! Wrap', wrapper);
    // console.log('!! HTML', wrapper.html());
    // console.log('!! EL', wrapper.element);

    // const vm = new Vue({
    //   el: document.createElement('div'),
    //   renderer: h => h(App),
    // });

    // renderer.renderToString(vm, (err, str) => {
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
