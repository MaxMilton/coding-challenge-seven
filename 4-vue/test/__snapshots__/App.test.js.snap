// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App root component renders correctly 1`] = `<!--function (a, b, c, d) { return createElement(vm, a, b, c, d, true); }-->`;
